llmfoundry/callbacks/curriculum_learning_callback.py

```diff
@@ -24,13 +21,18 @@
     BaseContextualError,
     TrainDataLoaderLocation,
 )
-from llmfoundry.utils.warnings import VersionedDeprecationWarning
 
 log = logging.getLogger(__name__)
 
 __all__ = ['CurriculumLearning']
 
 
+@dataclass
+class CurriculumLearningState:
+    schedule: list[dict[str, Any]]
```

Comment:
so as long as we use dataclass, the dataclass attributes also won't get improperly flattened right?

Reply:
Yes